packages/composables/src/composables/useFloating.ts

```diff
@@ -0,0 +1,226 @@
+import { FzFloatingPosition, FzUseFloatingArgs } from '../types'
+import { ref, reactive, onUnmounted } from 'vue'
+import { getHighestAvailableSpacePos } from '../utils'
+
+export const useFloating = (args: FzUseFloatingArgs) => {
+  const safeElementDomRef = ref<HTMLElement | null>(null)
+  const safeContainerDomRef = ref<HTMLElement | null>(null)
+  const safeOpenerDomRef = ref<HTMLElement | null>(null)
+  const rect = ref<DOMRect | null>(null)
+  const float = reactive({
+    position: { x: 0, y: 0 }
+  })
+  const options: IntersectionObserverInit = {
+    root: null,
+    rootMargin: '0px',
+    threshold: 1.0,
+    ...args.element.intersectionOptions
+  }
+
+  const handleIntersect = (
+    entries: IntersectionObserverEntry[],
+    observer: IntersectionObserver
+  ) => {}
+
+  const floatObserver = ref(new IntersectionObserver(handleIntersect, options))
+
+  const setPosition = () => {
+    safeElementDomRef.value =
+      typeof args.element.domRef.value === 'string'
+        ? document.querySelector(args.element.domRef.value)
+        : args.element.domRef.value
+    if (!args.container) {
+      safeContainerDomRef.value = document.body
+    } else {
+      safeContainerDomRef.value =
+        typeof args.container.domRef.value === 'string'
+          ? document.querySelector(args.container.domRef.value)
+          : args.container.domRef.value
+      safeContainerDomRef.value ??= document.body
+    }
+
+    if (!safeElementDomRef.value) {
+      throw new Error('missing reference element for floating behavior')
```

Comment:
is this catched anywhere? I would avoid throwing on the frontend unless strictly necessary, better yet to log on the console and eventually sentry